src/main/java/com/verisign/storm/metrics/reporters/KafkaReporter.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2014 VeriSign, Inc.
+ *
+ * VeriSign licenses this file to you under the Apache License, version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ *
+ * See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+ */
+package com.verisign.storm.metrics.reporters;
+
+import com.google.common.base.Throwables;
+import com.verisign.ie.styx.avro.graphingMetrics.GraphingMetrics;
+import com.verisign.storm.metrics.graphite.GraphiteCodec;
+import com.verisign.storm.metrics.graphite.GraphiteConnectionFailureException;
+import kafka.javaapi.producer.Producer;
+import kafka.producer.KeyedMessage;
+import kafka.producer.ProducerConfig;
+import org.apache.avro.Schema;
+import org.apache.avro.io.*;
+import org.apache.avro.specific.SpecificDatumReader;
+import org.apache.avro.specific.SpecificDatumWriter;
+import org.apache.avro.specific.SpecificRecordBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Properties;
+
+public class KafkaReporter extends AbstractReporter {
```

Comment:
Added in https://github.com/verisign/storm-graphite/commit/e37f962c05762a8be5c95252e8cd8358790a1579